cluster.go

```diff
@@ -423,6 +423,10 @@ func (c *Cluster) Shutdown() error {
 		} else {
 			time.Sleep(2 * time.Second)
 		}
+		/* set c.Config.Bootstrap to current peers */
+		c.config.Bootstrap = c.peerManager.peersAddrs()
+		c.config.Shadow()
```

Comment:
Hmm yeah, unconventional :)

There is a behaviour change here. Before, if the user passed in a flag like `--leave` but `leave_on_shutdown` was false in the configuration, the configuration would stay as false when the cluster is shudown. With this, however, it will switch to true when this code runs.

I think the cleanest here is to do `c.config.unshadow()` before editing the Bootstrap key. In both cases you are saving right afterwards, and that's when unshadow is supposed to happen anyway. Since you've done it before, whatever you write on `.Bootstrap` will stay.